### ISSUE TITLE
Infinite loop when constructor or destructor is defined with leading :: and a macro

### ISSUE THREAD

**Issue by alexhenrie** [uncrustify/uncrustify#3282]
Example code:

```
#include "foobar.hpp"

#define DESTRUCTOR Foobar::~Foobar

::DESTRUCTOR()
{
}
```

Uncrustify goes into an infinite loop when it encounters this code no matter what configuration is used.

**Comment by guy-maurel** [comment]
The current git version works pretty fine under Linux Mint.
  https://github.com/uncrustify/uncrustify
Please test it and report!
I get:
```.txt
Uncrustify_d-0.74.0-74-c89ce162

do_source_file(1505): Parsing: alexhenrie.cpp as language CPP
#include "foobar.hpp"

#define DESTRUCTOR Foobar::~Foobar

::DESTRUCTOR()
{
}

```

**Comment by alexhenrie** [comment]
It looks like this was fixed by 27a3bd915a6522b400960c26a0525a7794d38eaa, thanks!
